src/serac/physics/contact/contact_data.hpp

```diff
@@ -0,0 +1,202 @@
+// other Serac Project Developers. See the top-level LICENSE file for
+// details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+/**
+ * @file contact_data.hpp
+ *
+ * @brief Class for storing contact data
+ */
+
+#pragma once
+
+#include <mfem/fem/pgridfunc.hpp>
+#include "mfem.hpp"
+
+#include "serac/serac_config.hpp"
+#include "serac/physics/contact/contact_config.hpp"
+#include "serac/physics/state/finite_element_dual.hpp"
+#include "serac/physics/state/finite_element_state.hpp"
+#ifdef SERAC_USE_TRIBOL
+#include "serac/physics/contact/contact_pair.hpp"
+#endif
+
+namespace serac {
+
+namespace contact {
+
+/**
+ * @brief Default contact options: frictionless mortar with penalty = 1000
+ * enforcement
+ */
+const ContactOptions default_contact_options = {.method      = ContactMethod::SingleMortar,
+                                                .enforcement = ContactEnforcement::Penalty,
+                                                .type        = ContactType::Frictionless,
+                                                .penalty     = 1.0e3};
+
+}  // namespace contact
+
+/**
+ * @brief This class stores contact pair data and interacts with Tribol
+ */
+class ContactData {
+public:
+  /**
+   * @brief The constructor
+   *
+   * @param mesh The volume mesh for the problem
+   */
+  ContactData(const mfem::ParMesh& mesh);
+
+  /**
+   * @brief Destructor to finalize Tribol
+   */
+  ~ContactData();
+
+  /**
+   * @brief Add another contact pair
+   *
+   * @param pair_id Unique identifier for the ContactPair (used in Tribol)
+   * @param bdry_attr_surf1 MFEM boundary attributes for the first surface
+   * @param bdry_attr_surf2 MFEM boundary attributes for the second surface
+   * @param contact_opts Defines contact method, enforcement, type, and penalty
+   */
+  void addContactPair(int pair_id, const std::set<int>& bdry_attr_surf1, const std::set<int>& bdry_attr_surf2,
+                      ContactOptions contact_opts);
+
+  /**
+   * @brief Updates the positions, forces, and jacobian contributions associated with contact
+   *
+   * @param cycle The current simulation cycle
+   * @param time The current time
+   * @param dt The timestep size to attempt
+   */
+  void update(int cycle, double time, double& dt);
+
+  /**
+   * @brief Have there been contact pairs added?
+   *
+   * @return true if contact pairs have been added
+   * @return false if there are no contact pairs
+   */
+  bool haveContactPairs() const;
+
+  /**
+   * @brief Get the contact constraint residual (i.e. nodal forces)
+   *
+   * @return Nodal contact forces as a Vector
+   */
+  FiniteElementDual forces() const;
+
+  /**
+   * @brief Returns pressures on the contact surfaces
+   *
+   * @return Pressure finite element state
+   */
+  mfem::Vector mergedPressures() const;
+
+  /**
+   * @brief Returns nodal gaps on true degrees of freedom
+   *
+   * @return Nodal gaps as a Vector
+   */
+  mfem::Vector mergedGaps() const;
```

Comment:
what kind of mfem::Vector? T/L/E, ordering, polynomial order, etc

are the nodal gaps vector-valued or just the normal component?